packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanelMain.tsx

```diff
@@ -27,22 +27,39 @@ import { useSelectedToken } from '../../hooks/useSelectedToken'
 import { useBalances } from '../../hooks/useBalances'
 import { DestinationNetworkBox } from './TransferPanelMain/DestinationNetworkBox'
 import { SourceNetworkBox } from './TransferPanelMain/SourceNetworkBox'
-import { useArbQueryParams } from '../../hooks/useArbQueryParams'
+import {
+  DisabledFeatures,
+  useArbQueryParams
+} from '../../hooks/useArbQueryParams'
 import { addressesEqual } from '../../util/AddressUtils'
 import { CustomMainnetChainWarning } from './CustomMainnetChainWarning'
 import { getOrbitChains } from '../../util/orbitChainsList'
 import { useMode } from '../../hooks/useMode'
+import { useDisabledFeatures } from '../../hooks/useDisabledFeatures'
 
 export function SwitchNetworksButton(
   props: React.ButtonHTMLAttributes<HTMLButtonElement>
 ) {
   const { isSmartContractWallet, isLoading: isLoadingAccountType } =
     useAccountType()
 
-  const disabled = isSmartContractWallet || isLoadingAccountType
-
   const [networks, setNetworks] = useNetworks()
 
+  const { isFeatureDisabled } = useDisabledFeatures()
+
+  const disableTransfersToNonArbitrumChains = isFeatureDisabled(
+    DisabledFeatures.TRANSFERS_TO_NON_ARBITRUM_CHAINS
+  )
+
+  const destinationChainNonArbitrumNotAllowed =
```

Comment:
This isn't semantically correct, because `destinationChainNonArbitrumNotAllowed` (according the the variable name) should only tell us if non-Arbitrum chains can be used as the destination. But in fact we are checking for whether we can swap chains.

Maybe something like: `isSwapToNonArbitrumBlocked`